### PR TITLE
update to netcore

### PR DESCRIPTION
**Pull request by weitaolee**
update to net core 1.1

### CODE REVIEW

File: demo/DemoClient.csproj [hunk 1]
```diff
@@ -1,72 +0,0 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (moodmosaic) [inline demo/DemoClient.csproj]: Why is this file removed?

File: demo/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -1,42 +0,0 @@
-﻿// Copyright (c) 2014 - 2016 George Kimionis
```
Comment (moodmosaic) [inline demo/Properties/AssemblyInfo.cs]: Why is this file removed?

File: demo/packages.config [hunk 3]
```diff
@@ -1,5 +0,0 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (moodmosaic) [inline demo/packages.config]: Why is this file removed?

File: nuget/Build.config [hunk 4]
```diff
@@ -1,13 +0,0 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (moodmosaic) [inline nuget/Build.config]: Why is this file removed?

File: nuget/BuildRelease.msbuild [hunk 5]
```diff
@@ -1,76 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
```
Comment (moodmosaic) [inline nuget/BuildRelease.msbuild]: Why is this file removed?

File: nuget/NuGetSpecs/BitcoinLib.nuspec [hunk 6]
```diff
@@ -1,22 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
```
Comment (moodmosaic) [inline nuget/NuGetSpecs/BitcoinLib.nuspec]: Why is this file removed?

File: nuget/NuGetSpecs/Install.ps1 [hunk 7]
```diff
@@ -1,5 +0,0 @@
-param($installPath, $toolsPath, $package, $project)
```
Comment (moodmosaic) [inline nuget/NuGetSpecs/Install.ps1]: Why is this file removed?

File: nuget/build-release.sh [hunk 8]
```diff
@@ -1,30 +0,0 @@
-#!/bin/sh
```
Comment (moodmosaic) [inline nuget/build-release.sh]: Why is this file removed?

File: nuget/packages/Newtonsoft.Json.6.0.3/tools/install.ps1 [hunk 9]
```diff
@@ -1,93 +0,0 @@
-param($installPath, $toolsPath, $package, $project)
```
Comment (moodmosaic) [inline nuget/packages/Newtonsoft.Json.6.0.3/tools/install.ps1]: Why is this file removed?

File: nuget/packages/repositories.config [hunk 10]
```diff
@@ -1,4 +0,0 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (moodmosaic) [inline nuget/packages/repositories.config]: Why is this file removed?

File: src/BitcoinLib/BitcoinLib.csproj [hunk 11]
```diff
@@ -1,145 +0,0 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (moodmosaic) [inline src/BitcoinLib/BitcoinLib.csproj]: Why is this file removed?

File: src/BitcoinLib/RPC/Connector/IRpcConnector.cs [hunk 12]
```diff
@@ -1,12 +1,13 @@
...
 
+using System.Threading.Tasks;
 using BitcoinLib.RPC.Specifications;
 
 namespace BitcoinLib.RPC.Connector
 {
     public interface IRpcConnector
     {
-        T MakeRequest<T>(RpcMethods method, params object[] parameters);
+        Task<T> MakeRequestAsync<T>(RpcMethods method, params object[] parameters);
```
Comment (moodmosaic) [inline src/BitcoinLib/RPC/Connector/IRpcConnector.cs]: Isn't it a breaking change if we remove the 

```
T MakeRequest<T>(RpcMethods method, params object[] parameters)
```

call from the interface?

Comment (moodmosaic) [inline src/BitcoinLib/RPC/Connector/IRpcConnector.cs]: Is it *required* by .NET Core to have an async version of that call? Wouldn't it be easier if it's done on a separate pull request?

File: src/BitcoinLib/RPC/Connector/RawRpcConnector.cs [hunk 13]
```diff
@@ -1,49 +1,44 @@
...
         //  Usage example:  String networkDifficultyJsonResult = RawRpcConnector.MakeRequest("{\"method\":\"getdifficulty\",\"params\":[],\"id\":1}", new BitcoinService());
-        public static string MakeRequest(string jsonRequest, ICoinService coinService)
-        {
-            return MakeRequest(jsonRequest, coinService.Parameters.SelectedDaemonUrl, coinService.Parameters.RpcUsername, coinService.Parameters.RpcPassword);
-        }
+        //public static string MakeRequest(string jsonRequest, ICoinService coinService)
+        //{
+        //    return MakeRequest(jsonRequest, coinService.Parameters.SelectedDaemonUrl, coinService.Parameters.RpcUsername, coinService.Parameters.RpcPassword);
+        //}
     }
```
Comment (moodmosaic) [inline src/BitcoinLib/RPC/Connector/RawRpcConnector.cs]: Aren't we breaking stuff by commenting-out these lines of code?!

File: src/BitcoinLib/Responses/DecodeRawTransactionResponse.cs [hunk 14]
```diff
@@ -11,8 +10,8 @@ public class DecodeRawTransactionResponse : ITransactionResponse
     {
         public string Version { get; set; }
         public string LockTime { get; set; }
-        public List<Vin> Vin { get; set; }
-        public List<Vout> Vout { get; set; }
+        public Vin[] Vin { get; set; }
+        public Vout[] Vout { get; set; }
```
Comment (moodmosaic) [inline src/BitcoinLib/Responses/DecodeRawTransactionResponse.cs]: Isn't it a breaking change if we change these from `List<T>` to arrays?

File: src/BitcoinLib/Responses/GetAddedNodeInfoResponse.cs [hunk 15]
```diff
@@ -1,15 +1,14 @@
...
-using System.Collections.Generic;
 
 namespace BitcoinLib.Responses
 {
     public class GetAddedNodeInfoResponse
     {
         public string AddedNode { get; set; }
         public bool Connected { get; set; }
-        public List<NodeAddress> Addresses { get; set; }
+        public NodeAddress[] Addresses { get; set; }
```
Comment (moodmosaic) [inline src/BitcoinLib/Responses/GetAddedNodeInfoResponse.cs]: Isn't it a breaking change if we change these from `List<T>` to arrays?

File: src/BitcoinLib/Responses/GetBlockResponse.cs [hunk 16]
```diff
@@ -9,10 +9,10 @@ public class GetBlockResponse
     {
         public GetBlockResponse()
         {
-            Tx = new List<string>();
+            Tx = new List<string>().ToArray();
         }
 
-        public List<string> Tx { get; set; }
+        public string[] Tx { get; set; }
```
Comment (moodmosaic) [inline src/BitcoinLib/Responses/GetBlockResponse.cs]: Isn't it a breaking change if we change these from `List<T>` to arrays?

File: src/BitcoinLib/Responses/GetBlockTemplateResponse.cs [hunk 17]
```diff
@@ -1,20 +1,19 @@
...
         public int Version { get; set; }
         public string PreviousBlockHash { get; set; }
-        public List<GetBlockTemplateTransaction> Transactions { get; set; }
+        public GetBlockTemplateTransaction[] Transactions { get; set; }
         public GetBlockTemplateCoinbaseAux CoinbaseAux { get; set; }
         public long CoinbaseValue { get; set; }
         public string Target { get; set; }
         public int MinTime { get; set; }
-        public List<string> Mutable { get; set; }
+        public string[] Mutable { get; set; }
```
Comment (moodmosaic) [inline src/BitcoinLib/Responses/GetBlockTemplateResponse.cs]: Isn't it a breaking change if we change these from `List<T>` to arrays?

File: src/BitcoinLib/Responses/GetBlockTemplateResponse.cs [hunk 18]
```diff
@@ -1,20 +1,19 @@
...
-using System.Collections.Generic;
 
 namespace BitcoinLib.Responses
 {
     public class GetBlockTemplateResponse
     {
         public int Version { get; set; }
         public string PreviousBlockHash { get; set; }
-        public List<GetBlockTemplateTransaction> Transactions { get; set; }
+        public GetBlockTemplateTransaction[] Transactions { get; set; }
```
Comment (moodmosaic) [inline src/BitcoinLib/Responses/GetBlockTemplateResponse.cs]: Isn't it a breaking change if we change these from `List<T>` to arrays?

File: src/BitcoinLib/Responses/GetBlockTemplateResponse.cs [hunk 19]
```diff
@@ -32,7 +31,7 @@ public class GetBlockTemplateTransaction
     {
         public string Data { get; set; }
         public string Hash { get; set; }
-        public List<string> Depends { get; set; }
+        public string[] Depends { get; set; }
```
Comment (moodmosaic) [inline src/BitcoinLib/Responses/GetBlockTemplateResponse.cs]: Isn't it a breaking change if we change these from `List<T>` to arrays?

**Comment by GeorgeKimionis** [comment]
I totally agree with @moodmosaic's remarks. Switching to .Net Core is necessary, however this PR cannot be merged unless all aforementioned issues have been taken care of.
